packages/react/src/hooks/useClerkSignal.ts

```diff
@@ -0,0 +1,55 @@
+import type { SignInResource } from '@clerk/types';
+import { useCallback, useSyncExternalStore } from 'react';
+
+import { useIsomorphicClerkContext } from '../contexts/IsomorphicClerkContext';
+import { useAssertWrappedByClerkProvider } from './useAssertWrappedByClerkProvider';
+
+function useClerkSignal(signal: 'signIn'): { isLoaded: boolean; signIn: SignInResource | null } {
```

Comment:
Ditto, it feels that this should be able to accept any resource instead. Happy to wait for a pattern to emerge howerver